pitest/src/main/java/org/pitest/junit/JUnitTestClassIdentifier.java

```diff
@@ -39,7 +42,16 @@ public boolean isATestClass(final ClassInfo a) {
 
   @Override
   public boolean isIncluded(ClassInfo a) {
-    return isIncludedCategory(a) && !isExcludedCategory(a);
+    return isIncludedCategory(a) && !isExcludedCategory(a) && isNotRanWithExcludedRunner(a);
+  }
+
+  private boolean isNotRanWithExcludedRunner(ClassInfo a) {
+    String runWith = getRunWithAnnotationValue(a);
+    return this.excludedRunners.isEmpty() || !this.excludedRunners.contains(runWith);
```

Comment:
To be honest, I don't really know why I put that there. Probably it was part of my previous implementation. :-)

Gonna remove it.